libs/sdk-core/src/persist/transactions.rs

```diff
@@ -445,6 +447,13 @@ fn filter_to_where_clause(
         });
     }
 
+    if let Some(true) = is_jit {
+        where_clause.push(format!(
+            "payment_type = '{}' AND payer_amount_msat IS NOT NULL",
+            PaymentType::Received
+        ));
+    }
+
```

Comment:
@ok300 Yes, the user would expect to match either `Sent` and `ReceivedJIT`, though due to the above code snippet, the only results returned will be `ReceivedJIT` (since `payer_amount_msat` can only be not-null in received payments). 
If we really want to opt for this approach, we could change the payment-type filter clause from `payment_type in (<type1> <type2>)`  to an `OR` sequence: 
```sql
payment_type = <type1> OR payment_type = <type2> ... OR payment_type = ReceivedJIT and payer_amount_msat IS NOT NULL
```
But although I considered it I'm not sure whether it's the best approach.

Reply:
Oh I see, thanks.

Yes I think the `OR` sequence is a good idea, I'll explain below.

From what I can tell, the current version can lead to
```sql
WHERE
  # payment filters
  payment_type IN (A, B, C) AND
  # jit clauses
  payment_type = 'Received' AND payer_amount_msat IS NOT NULL
```

If the user doesn't include `Received` as a filter, but sets `is_jit = true`, the query will fail, because all these checks are chained via `AND`.

This wouldn't be a problem with your idea above, of chaining the `where` clauses in an `OR` sequence.

Besides, it even allows for paranthesis, which makes the query clearer

```sql
payment_type = <type1> OR
  payment_type = <type2> OR
  (payment_type = ReceivedJIT and payer_amount_msat IS NOT NULL)
```

I'd suggest going for this. Its clearer for the user (jit is just another filter in the list) and for us (better query structure).

Reply:
There is one thing that is a bit weird with this approach. We have the payment_type field on the Payment structure and so if you query for `ReceivedJIT` you will get payment with type `Received`...

Reply:
@roeierez Shouldn't that be intended? A received-jit is just a subset of a received payment, so it should be intuitive that it be of type received

Reply:
@ok300 I have implemented the `OR` approach, there shouldn't be any issues there. The code actually turned out pretty readable, and performance wise it may only be slightly worse.